api/books/book.service.ts

```diff
@@ -0,0 +1,84 @@
+import {Logger} from "../shared/logger.service";
+import {DEFAULT_REQUEST_DETAILS} from "../../koa.config";
+import {Book} from "./book.model";
+import {ApiError} from "../errors/api-error.errors";
+import {errors} from "../errors/errors";
+
+class BookService {
+
+    async create(bookData, details = DEFAULT_REQUEST_DETAILS) {
+        let book = new Book(bookData);
+        let savedBook = await book.save();
+        Logger.log("info", "[BookService] [save]");
+        let bookJson = savedBook.toJSON();
+        Logger.log("info", "[BookService] [Create] book created successfully", {book: bookJson, details});
+        return bookJson;
+    }
+
+    async findByIsbn(isbn, details = DEFAULT_REQUEST_DETAILS) {
+        let book = await Book.findOne({isbn: isbn});
+        if (!book) {
+            return null;
+        }
+
+        if (book && book.deletedAt) {
+            throw new ApiError(errors.generic.not_found);
+        }
+
+        Logger.log("info", "[BookService] [Find By Isbn] book retrieved successfully", {isbn, details});
+        return book.toJSON();
+    }
+
+    async search(query: any = {}, details = DEFAULT_REQUEST_DETAILS) {
+        let search: any = {};
+
+        if (query.q) {
+            search.$or = [
+                {"title": {$regex: query.q, $options: "i"}},
+                {"isbn": {$regex: query.q, $options: "i"}},
+            ];
+        }
+        if (query.active === undefined) {
+            query.active = "true";
```

Comment:
In general, it is not a good idea to modify an input parameter. For example, who calls _search_ does not know that _query_ parameter will change.